class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,111 @@
-    int lineNumber = 0;
-    bool found = false;
-    std::string lineFound;
 
-    std::vector<std::string> lines;
- 
-    while (std::getline(file, line)) {
-        lines.push_back(line);
+    int UserIncrement(const std::string& nameToFind) {
+         statsData[nameToFind]++;
+         SaveData();
+         return statsData[nameToFind];
     }
-    file.close();
-    for (const std::string& line : lines) {
-        lineNumber++;
-        if (line != "") {
-            if (splitString(line, ' ')[0] == WordToFind)
-            {
-                found = true;
-                lineFound = line;
-                break;
-            }
-        }
+    bool checkName(const std::string& nameToFind){
+        return statsData.find(nameToFind)!=statsData.end();
     }
-    if (!found) {
-        if (argument2 == "delete") {
-            std::cout << "There is no such user " <<std::endl;
-            return 0;
+private:
+    std::map<std::string, int> statsData;
+    const std::string databasePath;
+    void LoadData() {
+        std::ifstream file(databasePath);
+        std::string name;
+        int count;
+        while (file >> name >> count) {
+            statsData [name] = count;
         }
-        std::cout << "Welcome " << WordToFind << std::endl;
-        std::ofstream fileEdit("database.txt", std::ios::app);
-        if (!fileEdit.is_open()) {
-            std::cerr << "Could not open the file for writing!" << std::endl;
-            return 1;
+        file.close();
+    }
+    void SaveData() {
+        std::ofstream file(databasePath, std::ios::trunc);
+        for (const auto& [name, count] : statsData ) {
+            file << name << " " << count << std::endl;
         }
-        fileEdit << WordToFind << " 1" << std::endl;
-        fileEdit.close();
+        file.close();
     }
-    else {
-        if (argument2 == "delete") {
+};
 
-            for (int i = 0; i < lines.size(); i++)
-            {
-                if (splitString(lines[i], ' ')[0] == WordToFind)
-                {
-                    lines.erase(lines.begin() + i);
-                }
-            }
-            std::ofstream fileEdit("database.txt", std::ios::trunc);
-            for (const auto& line : lines) {
-                if (line != "")
-                {
-                    fileEdit << line << std::endl;
-                }
-            }
-            std::cout << "All data about this user was cleared" << std::endl;
+int notFoundNameHandling(const std::string& additionalCommand, const std::string& nameToFind, Database& db ){
+    if (additionalCommand == deleteCommand) {
+        std::println("There is no such user ");
+        return 1;
+    }
+    db.UserIncrement(nameToFind);
+    std::println("Welcome {}",nameToFind);
+    return 0;
+}
+
+int FoundNameHandling(const std::string& additionalCommand, const std::string& nameToFind, Database& db ){
+    if (additionalCommand == deleteCommand) {
+        if (db.deleteUser(nameToFind)==true){
```

Comment:
there is no need to compare bool with true. Just` if (db.deleteUser(nameToFind))`